### PR TITLE
Adding config files to onboard gitops-operator to OpenShift CI

### PR DESCRIPTION
**Pull request by dewan-ahmed**
Onboarding request for `redhat-developer/gitops-operator` to OpenShift CI

### CODE REVIEW

File: ci-operator/config/redhat-developer/gitops-operator/redhat-developer-gitops-operator-master.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  namespace: ocp
+tests:
+- as: e2e-aws
+  commands: TEST_SUITE=openshift/conformance/parallel run-tests
+  openshift_installer:
+    cluster_profile: aws
+- as: e2e-gitops-operator
+  commands: make test-e2e
+  openshift_installer_src:
+    cluster_profile: aws
```
Comment (amitkrout) [inline ci-operator/config/redhat-developer/gitops-operator/redhat-developer-gitops-operator-master.yaml:27]: @dewan-ahmed Can you please replace the line from 1-27 with
```yaml
build_root:
  project_image:
    dockerfile_path: openshift-ci/build-root/Dockerfile
resources:
  '*':
    requests:
      cpu: "2"
      memory: 4Gi
tests:
- artifact_dir: /tmp/artifacts
  as: unit
  commands: echo "Call unit test script"
  container:
    from: src
```

**Review by amitkrout — CHANGES_REQUESTED** [review]

File: ci-operator/config/redhat-developer/gitops-operator/redhat-developer-gitops-operator-master.yaml (hunk 1 above)
Comment (amitkrout) [reply ci-operator/config/redhat-developer/gitops-operator/redhat-developer-gitops-operator-master.yaml:27]: In the next pr you can go for spinning up the cluster. For now lets onboard the bare minimum openshift CI work flow.

File: core-services/prow/02_config/_config.yaml [hunk 2]
```diff
@@ -1821,6 +1821,7 @@ tide:
     - openshift/kubernetes
     - openshift/okd-machine-os
     - openshift/origin
+    - redhat-developer/gitops-operator
```
Comment (stevekuznetsov) [inline core-services/prow/02_config/_config.yaml:1824]: Move this line to be next to other `redhat-developer` repos

File: core-services/prow/02_config/_config.yaml [hunk 3]
```diff
@@ -1850,6 +1851,7 @@ tide:
     - openshift/ocs-operator
     - openshift/okd-machine-os
     - openshift/origin
+    - redhat-developer/gitops-operator
```
Comment (stevekuznetsov) [inline core-services/prow/02_config/_config.yaml:1854]: Move this line to be next to other `redhat-developer` repos

File: ci-operator/config/redhat-developer/gitops-operator/redhat-developer-gitops-operator-master.yaml (hunk 1 above)
Comment (dewan-ahmed) [reply ci-operator/config/redhat-developer/gitops-operator/redhat-developer-gitops-operator-master.yaml]: @amitkrout thanks. Review feedback implemented via 697be347f3b1d96b0e0d1abdb2f1b28d147c2cc1.

File: core-services/prow/02_config/_config.yaml (hunk 2 above)
Comment (dewan-ahmed) [reply core-services/prow/02_config/_config.yaml]: @stevekuznetsov thanks. Review feedback implemented via 697be347f3b1d96b0e0d1abdb2f1b28d147c2cc1.

File: core-services/prow/02_config/_config.yaml (hunk 3 above)
Comment (dewan-ahmed) [reply core-services/prow/02_config/_config.yaml]: @stevekuznetsov thanks. Review feedback implemented via 697be347f3b1d96b0e0d1abdb2f1b28d147c2cc1.

File: ci-operator/jobs/redhat-developer/gitops-operator/redhat-developer-gitops-operator-master-presubmits.yaml [hunk 4]
```diff
@@ -0,0 +1,156 @@
+presubmits:
```
Comment (amitkrout) [inline ci-operator/jobs/redhat-developer/gitops-operator/redhat-developer-gitops-operator-master-presubmits.yaml:1]: @dewan-ahmed Please run `make jobs` and commit the changes.

NOTE: Whenever you you do any changes in the config file, you need to generate it job file by running `make jobs`

**Review by amitkrout — CHANGES_REQUESTED** [review]

**Comment by dewan-ahmed** [comment]
Closing this PR since a new PR https://github.com/openshift/release/pull/13724 is created for onboarding gitops-operator.

**Pull request closed without merging**